### ISSUE TITLE
Cannot actually enter "treasure room" in tutorial

### ISSUE THREAD

**Issue by Griatch** [evennia/evennia#215]
_From [jeremywo...@gmail.com](https://code.google.com/u/111679206525907597994/) on March 13, 2012 19:48:57_

A) What steps will reproduce the problem? 1. Load up the tutorial code.
2. Log in as a regular user to play the tutorial.
3. Walk to the obelisk.
4. "look obelisk". Notice the results.
5. Attempt to go solve the puzzle by entering the hinted at tomb. For example, if this is the following clue:

The surface of the obelisk seem to waver, shift and writhe under your gaze, with different scenes and structures appearing whenever you look at it. You for a moment see the visage of a 

woman on a horse.

Then go to the tombs and type:

horse B) What is the expected output? What do you see instead (tracebacks, error messages)? I've been testing this more as an excuse to get used to the commands, but unfortunately I'm still too much of a noob at the code to figure out what's going on. What I expect is to enter the tomb and to not be sent to the failure room. Time and again when I try this, I have 100&#37; of the time been sent to the failure room. i've never made it to the treasure room.

C) What version/svn revision of Evennia are you using?
Tips of the code. D) Please provide any additional information/suggestions below. If I could figure out where the error was occurring, I'd submit a patch, but I'm a bit lost. I'll share what I was looking through.

I started in the TeleportRoom class that is part of contrib/tutorial_world/rooms.py file. I dropped some debug statements into the at_object_receive of the TeleportRoom and noticed that this function is always being called twice. Sometimes, interestingly, the following attributes are correct:
# In other words, this following test is true

character.db.puzzle_clue == self.db.puzzle_value

However, every time this method gets called a second time, the attributes are never equal.

I'm a bit confused:
- Why this method would be called more than once.
- Also, I'm still learning the code so I'm also concerned that I might be chasing the bug incorrectly.

I'm way more interested in going through this process to the learn the code, but I wanted to report this in case I was unable to figure things out (which, at the moment, is still the case).

_Original issue: http://code.google.com/p/evennia/issues/detail?id=215_
